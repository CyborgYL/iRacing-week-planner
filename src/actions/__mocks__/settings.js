import debounce from 'lodash.debounce';

export const getSettingsFromFirebase = jest.fn(() => ({ type: 'SETTINGS/LOAD_SETTINGS_FROM_FIREBASE' }));
export const saveSettingsToFirebase = jest.fn(() => ({ type: 'SETTINGS/FIREBASE_SYNCED' }));

const dispatchSaveSettingsToFirebase = (dispatch) => {
  dispatch(saveSettingsToFirebase());
};

export const debouncedDispatcherSaveSettings = debounce(dispatchSaveSettingsToFirebase, 10000);
