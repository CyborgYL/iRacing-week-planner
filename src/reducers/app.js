import moment, { duration } from 'moment';
import { CHANGE_MODAL, UPDATE_DAYS } from '../actions/app';
import { SIGNED_IN } from '../actions/auth';
import { seasonStart, weekSeasonStart } from '../config';

const currentDate = moment(new Date()).utc().startOf('day');
const currentDays = currentDate.diff(seasonStart, 'days');
const getWeek = (date) => Math.ceil(duration(moment(date).add({ second: 1 }).diff(weekSeasonStart)).asWeeks());

export default function app(
  state = {
    date: currentDate,
    daysSinceSeasonStart: currentDays,
    week: getWeek(currentDate),
    currentModal: null,
  },
  { type, days, modalName },
) {
  if (type === UPDATE_DAYS) {
    const date = moment(seasonStart).add(days, 'days');
    return {
      ...state,
      date,
      daysSinceSeasonStart: days,
      week: getWeek(date),
    };
  }

  if (type === CHANGE_MODAL) {
    return {
      ...state,
      currentModal: modalName,
    };
  }

  if (type === SIGNED_IN) {
    return {
      ...state,
      currentModal: null,
    };
  }

  return state;
}
