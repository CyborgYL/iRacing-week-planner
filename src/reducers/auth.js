// @flow

import { initializeApp } from 'firebase/app';
import type { FirebaseApp } from 'firebase/app';

import {
  LOADING_AUTH, SIGNED_OUT, SIGNED_IN, LOADING_RESET, RESET_SENT, ERROR_AUTH, ERROR_ACKNOWLEDGE, ERROR_RESET,
} from '../actions/auth';
import { firebaseConfig } from '../config';

const firebaseApp = initializeApp(firebaseConfig);

type AuthUser = {

};

type AuthState = {
  user: AuthUser,
  firebaseApp: FirebaseApp,
  loadingAuth: boolean,
  loadingReset: boolean,
  errorAuth: { message: string } | null,
  errorReset: { message: string } | null,
};

type AuthReducerArgs = {
  type: string,
  user?: AuthUser,
  error?: { message: string },
};

export default function authReducer(
  state: AuthState = {
    user: null,
    firebaseApp,
    loadingAuth: true,
    loadingReset: false,
    errorAuth: null,
    errorReset: null,
  },
  { type, user, error }: AuthReducerArgs,
): AuthState {
  if (type === SIGNED_OUT) {
    return { ...state, user: null };
  }

  if (type === LOADING_AUTH) {
    return { ...state, loadingAuth: true };
  }

  if (type === ERROR_AUTH) {
    return { ...state, loadingAuth: false, errorAuth: error };
  }

  if (type === ERROR_RESET) {
    return { ...state, loadingReset: false, errorReset: error };
  }

  if (type === ERROR_ACKNOWLEDGE) {
    return { ...state, errorAuth: null, errorReset: null };
  }

  if (type === SIGNED_IN) {
    return { ...state, loadingAuth: false, errorAuth: null, user };
  }

  if (type === LOADING_RESET) {
    return { ...state, loadingReset: true };
  }

  if (type === RESET_SENT) {
    return { ...state, loadingReset: false, errorReset: null };
  }

  return state;
}
