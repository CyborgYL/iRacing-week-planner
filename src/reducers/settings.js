// @flow

import uniq from 'lodash.uniq';
import { actionTypes as localStorageActionTypes } from 'redux-localstorage';
import { SIGNED_OUT } from '../actions/auth';
import {
  FIREBASE_SYNCED,
  LOAD_SETTINGS_FROM_FIREBASE,
  RESET_FILTERS,
  RESET_SETTINGS,
  UPDATE_FILTERS,
  UPDATE_SETTING,
} from '../actions/settings';

import availableColumns from '../data/availableColumns';
import { tracks, cars } from '../data';

export type sort = {
  key: string,
  order: string,
};

export type filters = {
  type: Array<string>,
  licence: Array<string>,
  official: Array<boolean>,
  fixed: boolean[],
  ownedCars: boolean,
  ownedTracks: boolean,
  favouriteSeries: boolean,
  favouriteCarsOnly: boolean,
  favouriteTracksOnly: boolean,
};

export const defaultFilters: filters = {
  type: ['Road', 'Oval', 'Dirt', 'RX'],
  licence: ['R', 'D', 'C', 'B', 'A', 'P'],
  official: [false, true],
  fixed: [false, true],
  ownedCars: false,
  ownedTracks: false,
  favouriteSeries: false,
  favouriteTracksOnly: false,
  favouriteCarsOnly: false,
};

export const defaultSettings = {
  filters: defaultFilters,
  ownedCars: cars.filter((car) => car.freeWithSubscription === true).map((car) => car.sku),
  ownedTracks: tracks.filter((track) => track.default).map((track) => track.pkgid),
  favouriteSeries: [],
  favouriteCars: [],
  favouriteTracks: [],
  sort: { key: 'licence', order: 'asc' },
  columns: availableColumns.filter((column) => column.default === true).map((column) => column.id),
  firebaseSynced: false,
};

const LEGACY_STORAGE_KEY = 'iracing-state';

const legacyStored = window.localStorage.getItem(LEGACY_STORAGE_KEY);
const startupSettings = legacyStored ? JSON.parse(legacyStored) : defaultSettings;

export default function settings(state = startupSettings, { type, payload }) {
  if (type === localStorageActionTypes.INIT) {
    if (!payload) {
      return state;
    }

    const persistedStateSettings = { ...payload.settings };

    persistedStateSettings.ownedCars = uniq([
      ...persistedStateSettings.ownedCars,
      ...defaultSettings.ownedCars,
    ]);

    persistedStateSettings.ownedTracks = uniq([
      ...persistedStateSettings.ownedTracks,
      ...defaultSettings.ownedTracks,
    ]);

    window.localStorage.removeItem(LEGACY_STORAGE_KEY);

    return { ...state, ...persistedStateSettings };
  }

  if (type === UPDATE_FILTERS) {
    return { ...state, filters: payload.filters, firebaseSynced: false };
  }

  if (type === RESET_FILTERS) {
    return { ...state, filters: defaultFilters, firebaseSynced: false };
  }

  if (type === RESET_SETTINGS || type === SIGNED_OUT) {
    return { ...defaultSettings, firebaseSynced: false };
  }

  if (type === UPDATE_SETTING) {
    return { ...state, [payload.key]: payload.value, firebaseSynced: false };
  }

  if (type === LOAD_SETTINGS_FROM_FIREBASE) {
    return { ...state, ...defaultSettings, ...payload, firebaseSynced: true };
  }

  if (type === FIREBASE_SYNCED) {
    return { ...state, firebaseSynced: true };
  }

  return state;
}
