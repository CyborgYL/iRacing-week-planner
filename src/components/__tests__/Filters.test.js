import { describe, test } from '@jest/globals';
import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import ConnectedFilters, { Filters } from '../Filters';
import { defaultFilters } from '../../reducers/settings'

const mockStore = configureMockStore();

describe('components/Filters', () => {
  const updateFilters = jest.fn();
  const resetSettings = jest.fn();
  const resetFilters = jest.fn();
  const t = jest.fn((key) => key);

  test('renders correctly', () => {
    const component = shallow(<Filters
      currentFilters={defaultFilters}
      updateFilters={updateFilters}
      resetSettings={resetSettings}
      resetFilters={resetFilters}
      t={t}
    />);

    expect(component).toMatchSnapshot();
  });

  test('correctly click a filter item checkbox', () => {
    const component = shallow(<Filters
      currentFilters={{ ...defaultFilters, type: ['Oval', 'Road'] }}
      updateFilters={updateFilters}
      resetSettings={resetSettings}
      resetFilters={resetFilters}
      t={t}
    />);

    const ovalCheckbox = component.find({ id: 'checkbox-type-oval' });

    ovalCheckbox.first().invoke('onChange')(false);

    expect(updateFilters).toHaveBeenCalledWith(expect.objectContaining({
      type: ['Road']
    }));

    updateFilters.mockClear();

    const rxCheckbox = component.find({ id: 'checkbox-type-rx' });

    rxCheckbox.first().invoke('onChange')(true);

    expect(updateFilters).toHaveBeenCalledWith(expect.objectContaining({
      type: ['Oval', 'Road', 'RX']
    }));

    expect(component).toMatchSnapshot();
  });

  test('correctly click a true/false checkbox', () => {
    const component = shallow(<Filters
      currentFilters={{ ...defaultFilters, ownedTracks: true }}
      updateFilters={updateFilters}
      resetSettings={resetSettings}
      resetFilters={resetFilters}
      t={t}
    />);

    const ownedTracksCheckbox = component.find({ id: 'checkbox-ownedTracks' });

    ownedTracksCheckbox.first().invoke('onChange')(false);

    expect(updateFilters).toHaveBeenCalledWith(expect.objectContaining({
      ownedTracks: false,
    }));

    updateFilters.mockClear();

    const favouriteSeriesCheckbox = component.find({ id: 'checkbox-favouriteSeries' });

    favouriteSeriesCheckbox.first().invoke('onChange')(true);

    expect(updateFilters).toHaveBeenCalledWith(expect.objectContaining({
      favouriteSeries: true,
    }));

    expect(component).toMatchSnapshot();
  });

  test('not synced user', () => {
    const component = shallow(<Filters
      currentFilters={defaultFilters}
      updateFilters={updateFilters}
      resetSettings={resetSettings}
      resetFilters={resetFilters}
      t={t}
      user={{ id: 12 }}
      firebaseSynced={false}
    />);

    expect(component).toMatchSnapshot();
  });

  test('synced user', () => {
    const component = shallow(<Filters
      currentFilters={defaultFilters}
      updateFilters={updateFilters}
      resetSettings={resetSettings}
      resetFilters={resetFilters}
      t={t}
      user={{ id: 12 }}
      firebaseSynced
    />);

    expect(component).toMatchSnapshot();
  });

  test('buttons fire events', () => {
    const component = shallow(<Filters
      currentFilters={defaultFilters}
      updateFilters={updateFilters}
      resetSettings={resetSettings}
      resetFilters={resetFilters}
      t={t}
      user={{ id: 12 }}
      firebaseSynced
    />);
    component.find({ id: 'filters-reset-filters-button' }).first().simulate('click');
    expect(resetFilters).toBeCalled();

    component.find({ id: 'filters-reset-settings-button' }).first().simulate('click');
    expect(resetSettings).toBeCalled();
  });

  test('passes redux props', () => {
    const store = mockStore({
      settings: { filters: defaultFilters, firebaseSynced: true },
      auth: { user: null },
    });
    const component = shallow(<ConnectedFilters store={store} />);
    expect(component.children().first().prop('currentFilters')).toBe(defaultFilters);
    expect(component.children().first().prop('user')).toBe(null);
    expect(component.children().first().prop('firebaseSynced')).toBe(true);
  });
});
