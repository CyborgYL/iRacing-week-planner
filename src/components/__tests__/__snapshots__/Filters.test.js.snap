// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Filters correctly click a filter item checkbox 1`] = `
<div
  style={
    Object {
      "fontSize": "0.8em",
    }
  }
>
  <h4>
    Type
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-oval"
    onChange={[Function]}
  >
    Oval
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-road"
    onChange={[Function]}
  >
    Road
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-type-dirt"
    onChange={[Function]}
  >
    Dirt
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-type-rx"
    onChange={[Function]}
  >
    RX
  </Checkbox>
  <h4>
    Licence
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-r"
    onChange={[Function]}
  >
    R
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-d"
    onChange={[Function]}
  >
    D
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-c"
    onChange={[Function]}
  >
    C
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-b"
    onChange={[Function]}
  >
    B
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-a"
    onChange={[Function]}
  >
    A
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-p"
    onChange={[Function]}
  >
    P
  </Checkbox>
  <h4>
    Official/Fixed
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-false"
    onChange={[Function]}
  >
    Unofficial
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-true"
    onChange={[Function]}
  >
    Official
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-false"
    onChange={[Function]}
  >
    Open setup
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-true"
    onChange={[Function]}
  >
    Fixed setup
  </Checkbox>
  <h4>
    Content
  </h4>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-ownedCars"
    onChange={[Function]}
  >
    Owned cars only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-ownedTracks"
    onChange={[Function]}
  >
    Owned tracks only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteSeries"
    onChange={[Function]}
  >
    Favorite series only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteCarsOnly"
    onChange={[Function]}
  >
    Favorite cars only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteTracksOnly"
    onChange={[Function]}
  >
    Favorite tracks only
  </Checkbox>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-filters-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset filters
    </button>
  </p>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-settings-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset all settings
    </button>
  </p>
</div>
`;

exports[`components/Filters correctly click a true/false checkbox 1`] = `
<div
  style={
    Object {
      "fontSize": "0.8em",
    }
  }
>
  <h4>
    Type
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-oval"
    onChange={[Function]}
  >
    Oval
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-road"
    onChange={[Function]}
  >
    Road
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-dirt"
    onChange={[Function]}
  >
    Dirt
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-rx"
    onChange={[Function]}
  >
    RX
  </Checkbox>
  <h4>
    Licence
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-r"
    onChange={[Function]}
  >
    R
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-d"
    onChange={[Function]}
  >
    D
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-c"
    onChange={[Function]}
  >
    C
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-b"
    onChange={[Function]}
  >
    B
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-a"
    onChange={[Function]}
  >
    A
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-p"
    onChange={[Function]}
  >
    P
  </Checkbox>
  <h4>
    Official/Fixed
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-false"
    onChange={[Function]}
  >
    Unofficial
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-true"
    onChange={[Function]}
  >
    Official
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-false"
    onChange={[Function]}
  >
    Open setup
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-true"
    onChange={[Function]}
  >
    Fixed setup
  </Checkbox>
  <h4>
    Content
  </h4>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-ownedCars"
    onChange={[Function]}
  >
    Owned cars only
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-ownedTracks"
    onChange={[Function]}
  >
    Owned tracks only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteSeries"
    onChange={[Function]}
  >
    Favorite series only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteCarsOnly"
    onChange={[Function]}
  >
    Favorite cars only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteTracksOnly"
    onChange={[Function]}
  >
    Favorite tracks only
  </Checkbox>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-filters-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset filters
    </button>
  </p>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-settings-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset all settings
    </button>
  </p>
</div>
`;

exports[`components/Filters not synced user 1`] = `
<div
  style={
    Object {
      "fontSize": "0.8em",
    }
  }
>
  <h4>
    Type
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-oval"
    onChange={[Function]}
  >
    Oval
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-road"
    onChange={[Function]}
  >
    Road
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-dirt"
    onChange={[Function]}
  >
    Dirt
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-rx"
    onChange={[Function]}
  >
    RX
  </Checkbox>
  <h4>
    Licence
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-r"
    onChange={[Function]}
  >
    R
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-d"
    onChange={[Function]}
  >
    D
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-c"
    onChange={[Function]}
  >
    C
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-b"
    onChange={[Function]}
  >
    B
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-a"
    onChange={[Function]}
  >
    A
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-p"
    onChange={[Function]}
  >
    P
  </Checkbox>
  <h4>
    Official/Fixed
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-false"
    onChange={[Function]}
  >
    Unofficial
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-true"
    onChange={[Function]}
  >
    Official
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-false"
    onChange={[Function]}
  >
    Open setup
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-true"
    onChange={[Function]}
  >
    Fixed setup
  </Checkbox>
  <h4>
    Content
  </h4>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-ownedCars"
    onChange={[Function]}
  >
    Owned cars only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-ownedTracks"
    onChange={[Function]}
  >
    Owned tracks only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteSeries"
    onChange={[Function]}
  >
    Favorite series only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteCarsOnly"
    onChange={[Function]}
  >
    Favorite cars only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteTracksOnly"
    onChange={[Function]}
  >
    Favorite tracks only
  </Checkbox>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-filters-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset filters
    </button>
  </p>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-settings-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset all settings
    </button>
  </p>
  <p>
    <span
      id="filters-synced-status"
    >
      Awaiting sync
      (refresh browser to download latest)
    </span>
  </p>
</div>
`;

exports[`components/Filters renders correctly 1`] = `
<div
  style={
    Object {
      "fontSize": "0.8em",
    }
  }
>
  <h4>
    Type
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-oval"
    onChange={[Function]}
  >
    Oval
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-road"
    onChange={[Function]}
  >
    Road
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-dirt"
    onChange={[Function]}
  >
    Dirt
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-rx"
    onChange={[Function]}
  >
    RX
  </Checkbox>
  <h4>
    Licence
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-r"
    onChange={[Function]}
  >
    R
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-d"
    onChange={[Function]}
  >
    D
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-c"
    onChange={[Function]}
  >
    C
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-b"
    onChange={[Function]}
  >
    B
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-a"
    onChange={[Function]}
  >
    A
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-p"
    onChange={[Function]}
  >
    P
  </Checkbox>
  <h4>
    Official/Fixed
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-false"
    onChange={[Function]}
  >
    Unofficial
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-true"
    onChange={[Function]}
  >
    Official
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-false"
    onChange={[Function]}
  >
    Open setup
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-true"
    onChange={[Function]}
  >
    Fixed setup
  </Checkbox>
  <h4>
    Content
  </h4>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-ownedCars"
    onChange={[Function]}
  >
    Owned cars only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-ownedTracks"
    onChange={[Function]}
  >
    Owned tracks only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteSeries"
    onChange={[Function]}
  >
    Favorite series only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteCarsOnly"
    onChange={[Function]}
  >
    Favorite cars only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteTracksOnly"
    onChange={[Function]}
  >
    Favorite tracks only
  </Checkbox>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-filters-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset filters
    </button>
  </p>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-settings-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset all settings
    </button>
  </p>
</div>
`;

exports[`components/Filters synced user 1`] = `
<div
  style={
    Object {
      "fontSize": "0.8em",
    }
  }
>
  <h4>
    Type
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-oval"
    onChange={[Function]}
  >
    Oval
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-road"
    onChange={[Function]}
  >
    Road
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-dirt"
    onChange={[Function]}
  >
    Dirt
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-type-rx"
    onChange={[Function]}
  >
    RX
  </Checkbox>
  <h4>
    Licence
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-r"
    onChange={[Function]}
  >
    R
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-d"
    onChange={[Function]}
  >
    D
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-c"
    onChange={[Function]}
  >
    C
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-b"
    onChange={[Function]}
  >
    B
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-a"
    onChange={[Function]}
  >
    A
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-licence-p"
    onChange={[Function]}
  >
    P
  </Checkbox>
  <h4>
    Official/Fixed
  </h4>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-false"
    onChange={[Function]}
  >
    Unofficial
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-official-true"
    onChange={[Function]}
  >
    Official
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-false"
    onChange={[Function]}
  >
    Open setup
  </Checkbox>
  <Checkbox
    checked={true}
    disabled={false}
    id="checkbox-fixed-true"
    onChange={[Function]}
  >
    Fixed setup
  </Checkbox>
  <h4>
    Content
  </h4>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-ownedCars"
    onChange={[Function]}
  >
    Owned cars only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-ownedTracks"
    onChange={[Function]}
  >
    Owned tracks only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteSeries"
    onChange={[Function]}
  >
    Favorite series only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteCarsOnly"
    onChange={[Function]}
  >
    Favorite cars only
  </Checkbox>
  <Checkbox
    checked={false}
    disabled={false}
    id="checkbox-favouriteTracksOnly"
    onChange={[Function]}
  >
    Favorite tracks only
  </Checkbox>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-filters-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset filters
    </button>
  </p>
  <p>
    <button
      className="btn btn-primary"
      id="filters-reset-settings-button"
      onClick={[MockFunction]}
      type="button"
    >
      Reset all settings
    </button>
  </p>
  <p>
    <span
      id="filters-synced-status"
    >
      Synced
      (refresh browser to download latest)
    </span>
  </p>
</div>
`;
